### DISCUSSION TITLE
How to debug custom queries not loading

### DISCUSSION THREAD

**Discussion by ayroblu** [nvim-treesitter/nvim-treesitter#6372] [Q&A] [answered]
I have a new textobjects.scm query that I'm trying to add.
My understanding is that I need to put it in ~/.config/nvim/queries/{lang}/textobjects.scm
However it doesn't work. But I don't know the first step to debug this kind of issue.

My rtp from `:lua print(vim.inspect(vim.api.nvim_list_runtime_paths()))` contains `~/.config/nvim` and other stuff like `init.lua` uses that so I think that seems right. My queries are fine cause if I copy paste them into the `~/.vim/plugged/nvim-treesitter-textobjects/queries/{lang}/textobjects.scm` file, it works just fine. It just seems like it's not loading my custom queries, but I don't know how to figure out why

Would love any debugging tips

**Comment by theHamsta** [comment]
It would be good to find out the `{lang}` that your file is using. It is not necessarily identical to the file type. For example, multiple file types can share a parser.

Which parser are you targeting? Do the built-in in text objects work (if existing)?

**Reply by ayroblu** [reply]
tsx, but I know lang is not the issue cause I can paste custom queries into the library file and it works, i.e. `~/.vim/plugged/nvim-treesitter-textobjects/queries/tsx/textobjects.scm`

**Comment by ayroblu** [comment, marked as answer]
My understanding from debugging after I realised that an invalid query in the lib folder would throw:

Code is here: https://github.com/neovim/neovim/blob/38e38d1b4/runtime/lua/vim/treesitter/query.lua#L94

```lua
local query_path = string.format('queries/%s/%s.scm', "tsx", "textobjects")
print(dump(vim.api.nvim_get_runtime_file(query_path, true)))
print(dump(vim.treesitter.query.get_files("tsx", "textobjects")))
```

First useful commands to see what is found, first print gets all the query files found (local config is found), second is what ends up being returned (doesn't have my file).

Then from reading the code, looks like it needs to have an extends on top, and just randomly tried, needs to be in the after folder
